x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -227,28 +391,21 @@ public static Authentication randomAuthentication(User user, RealmRef realmRef)
     }
 
     public static Authentication randomApiKeyAuthentication(User user, String apiKeyId) {
-        final RealmRef apiKeyRealm = new RealmRef("_es_api_key", "_es_api_key", randomAlphaOfLengthBetween(3, 8));
+        return randomApiKeyAuthentication(user, apiKeyId, VersionUtils.randomVersionBetween(random(), Version.V_7_0_0, Version.CURRENT));
+    }
+
+    public static Authentication randomApiKeyAuthentication(User user, String apiKeyId, Version version) {
         final HashMap<String, Object> metadata = new HashMap<>();
         metadata.put(AuthenticationField.API_KEY_ID_KEY, apiKeyId);
         metadata.put(AuthenticationField.API_KEY_NAME_KEY, randomBoolean() ? null : randomAlphaOfLengthBetween(1, 16));
-        return new Authentication(
-            user,
-            apiKeyRealm,
-            user.isRunAs() ? new RealmRef("lookup_realm", "lookup_realm", randomAlphaOfLength(5)) : null,
-            VersionUtils.randomVersionBetween(random(), Version.V_7_0_0, Version.CURRENT),
-            AuthenticationType.API_KEY,
-            metadata
-        );
+        return Authentication.newApiKeyAuthentication(AuthenticationResult.success(user, metadata), randomAlphaOfLengthBetween(3, 8))
+            .maybeRewriteForOlderVersion(version);
```

Comment:
We had a good discussion on this topic.
In principle, a node should construct authentications on the current version only, which are then downgraded automatically when serialized.

Reply:
The code won't throw NPE because the main code `maybeRewriteMetadataForApiKeyRoleDescriptors` is updated to check whether the keys exist. It's non-ideal to change main code for tests. But we agreed that I will follow up with a separate PR.